xtask/src/commands/release.rs

```diff
@@ -222,19 +168,17 @@ impl Prepare {
                 "apollo-router"
             ]),
             Version::Nightly => {
-                let head_commit: String = match Repository::open_from_env() {
-                    Ok(repo) => {
-                        let revspec = repo.revparse("HEAD")?;
-                        if revspec.mode().contains(git2::RevparseMode::SINGLE) {
-                            let mut full_hash = revspec.from().unwrap().id().to_string();
-                            full_hash.truncate(8);
-                            full_hash
-                        } else {
-                            panic!("unexpected rev-parse HEAD");
-                        }
-                    }
-                    Err(e) => panic!("failed to open: {e}"),
-                };
+                let head_commit: String = std::process::Command::new("git")
+                    .args(["rev-parse", "HEAD"])
+                    .output()
+                    .expect("failed to execute 'git rev-parse HEAD'")
+                    .stdout
+                    .iter()
+                    .map(|b| *b as char)
+                    .collect::<String>()
```

Comment:
This roughly decodes bytes to Unicode as ISO-8859-1. It shouldn’t make a difference if the output is ASCII, but why not `str::from_utf8` or `String::from_utf8_lossy`?

Reply:
I have spoken with the robot who wrote this code originally and we've determined there is a path forward, which has been demonstrated in 66913a2. 😄